tests/std/tests/P0429R9_flat_map/test.cpp

```diff
@@ -12,22 +12,25 @@
 
 using namespace std;
 
+template <class T, template <class...> class Tmpl>
+constexpr bool is_specialization_v = false;
+template <template <class...> class Tmpl, class... Ts>
+constexpr bool is_specialization_v<Tmpl<Ts...>, Tmpl> = true;
+
 template <typename T>
-concept IsFlatMap = _Is_specialization_v<remove_cvref_t<T&&>, flat_map>;
+concept IsFlatMap = is_specialization_v<remove_cvref_t<T>, flat_map>;
 
 template <IsFlatMap T>
 bool check_container_requirements(T&&) {
     return true;
 }
 
 template <IsFlatMap T>
-consteval bool check_reversible_container_requirements()
-    requires _Is_specialization_v<remove_cvref_t<T>, flat_map>
-{
+consteval bool check_reversible_container_requirements() {
     using map_t = remove_cvref_t<T>;
     bool result = true;
-    result &= is_same_v<std::reverse_iterator<typename map_t::iterator>, typename map_t::reverse_iterator>;
-    result &= is_same_v<std::reverse_iterator<typename map_t::const_iterator>, typename map_t::const_reverse_iterator>;
+    result &= is_same_v<reverse_iterator<typename map_t::iterator>, typename map_t::reverse_iterator>;

```

Comment:
Pre-existing: Applying `&=` to `bool` is squirrely and we don't do this in the codebase.